Request OPTIONS fix to return all allowed methods

michalbundyra opened the pull request:
The integration tests are pulled from zend-expressive-router repository.

See: https://github.com/zendframework/zend-expressive-router/pull/53

michalbundyra committed:
Fixed broken tests

Interface of zend-expressive-router has been changed with latest alpha
release

michalbundyra committed:
Added ImplicitOptionsMiddlewareIntegrationTest

Check if all allowed methods are included in the Allow header for OPTIONS
request

michalbundyra committed:
Matched route contains all allowed methods for OPTIONS requests

michalbundyra committed:
Updated year in license header

michalbundyra committed:
OPTIONS request returns failure and RouteResult contains allowed methods

michalbundyra committed:
Updated integration tests for HEAD and OPTIONS requests

michalbundyra committed:
Use const instead of strings - RequestMethods

michalbundyra committed:
Updated integration tests - explicit HEAD and OPTIONS requests

geerteltink commented on composer.json:
```diff
@@ -20,19 +20,26 @@
+    "repositories": [
+        {
+            "type": "git",
+            "url": "https://github.com/webimpress/zend-expressive-router"
```
A reminder to remove this before merging.

michalbundyra committed:
Updated IntegrationTest from zend-expressive-router (more assertions)

weierophinney committed:
Updates to zend-expressive-router 3.0.0rc1

Pins to 3.0.0rc1, and updates new test case based on name change in that
release.
